tests/testthat/test-connection.R

```diff
@@ -32,11 +32,6 @@ test_that("id() works", { for (conn in conns) { # nolint: brace_linter
   expect_identical(id("test_mtcars"), DBI::Id(table = "test_mtcars"))
   expect_identical(id("test.mtcars"), DBI::Id(schema = "test", table = "mtcars"))
 
-  if (inherits(conn, "SQLiteConnection")) {
-    expect_identical(id("test.mtcars", conn), DBI::Id(table = "test.mtcars"))
-  } else {
-    expect_identical(id("test.mtcars", conn), DBI::Id(schema = "test", table = "mtcars"))
-  }
 }})
```

Comment:
Here you could (should?) test that your schema detection heuristic works for sqlite.

That is for SQLite in this test setup we expect:

SCDB::id("test.mtcars", conn) == DBI::Id(schema = "test", table = "mtcars")
SCDB::id("not_test.mtcars", conn) == DBI::Id(table = "not_test.mtcars")

Reply:
That went a bit too fast, oops!

I have re-added the test for whether `id(test.mtcars)` and `DBI::Id(schema = "test", table = "mtcars")` are identical and a separate test for the `allow_table_only` argument.